File: logica/python/exercicio17.py
# Crie uma função em Python chamada 'inverter_lista' que receba uma lista como argumento.
#
# A função deve retornar uma NOVA lista com os elementos da lista original em ordem inversa.
#
# **Regras:**
# - Não use o método `.reverse()` ou `[::-1]` da lista. Faça a inversão usando um loop.


def inverter_lista(lista):
    invertida = []
    for i in range(len(lista) - 1, -1, -1):
        invertida.append(lista[i])
    return invertida

print(inverter_lista([1, 2, 3, 4, 5]))      
print(inverter_lista([52, 53, 54, 55, 56]))  

File: logica/javascript/exercicio17.js
//
// Crie uma função em JavaScript chamada 'calcularMediaNotas' que receba
// um array de números (representando notas) como argumento.
//
// A função deve retornar a média aritmética de todas as notas no array.
//
// **Regras:**
// - Se o array estiver vazio, retorne 0 para evitar divisão por zero.

function calcularMediaNotas(notas) {
    if (notas.length === 0) {
        return 0;
    }

    let soma = 0;
    for (let i = 0; i < notas.length; i++) {
        soma += notas[i];
    }

    const media = soma / notas.length;
    return media;
}

console.log(calcularMediaNotas([10, 20, 30]));
console.log(calcularMediaNotas([70, 80, 90, 100]));
console.log(calcularMediaNotas([]));
console.log(calcularMediaNotas([50]));